contracts/refs/CoreRef.sol

```diff
@@ -91,6 +91,27 @@ abstract contract CoreRef is ICoreRef, Pausable {
         _;
     }
 
+    // Modifiers to allow any combination of roles
+    modifier hasAnyOfTwoRoles(bytes32 role1, bytes32 role2) {
+        require(_core.hasRole(role1, msg.sender) || _core.hasRole(role2, msg.sender), "UNAUTHORIZED");
+        _;
+    }
+
+    modifier hasAnyOfThreeRoles(bytes32 role1, bytes32 role2, bytes32 role3) {
+        require(_core.hasRole(role1, msg.sender) || _core.hasRole(role2, msg.sender) || _core.hasRole(role3, msg.sender), "UNAUTHORIZED");
+        _;
+    }
+
+    modifier hasAnyOfFourRoles(bytes32 role1, bytes32 role2, bytes32 role3, bytes32 role4) {
+        require(_core.hasRole(role1, msg.sender) || _core.hasRole(role2, msg.sender) || _core.hasRole(role3, msg.sender) || _core.hasRole(role4, msg.sender), "UNAUTHORIZED");
+        _;
+    }
+
+    modifier hasAnyOfFiveRoles(bytes32 role1, bytes32 role2, bytes32 role3, bytes32 role4, bytes32 role5) {
+        require(_core.hasRole(role1, msg.sender) || _core.hasRole(role2, msg.sender) || _core.hasRole(role3, msg.sender) || _core.hasRole(role4, msg.sender) || _core.hasRole(role5, msg.sender), "UNAUTHORIZED");
+        _;
+    }
```

Comment:
Lets remove 5 until we need it? I think it would trim bytecode size but maybe compiler does that automatically

Reply:
Compiler does this automatically; modifiers are inlined into functions that use them, so if no functions use it it won't be in the bytecode